### PR TITLE
DTSPB-3524 upgrade java17, appinsights

### PR DESCRIPTION
**Pull request by FeliTam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-3524

### Change description ###
upgrade java 17, appinsights 3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-ARG APP_INSIGHTS_AGENT_VERSION=2.5.1
-FROM hmctspublic.azurecr.io/base/java:11-distroless
+ARG APP_INSIGHTS_AGENT_VERSION=3.2.10
```
Comment (timja) [inline Dockerfile]:
```suggestion
 # renovate: datasource=github-releases depName=microsoft/ApplicationInsights-Java
ARG APP_INSIGHTS_AGENT_VERSION=3.4.12
```

**Review by timja — COMMENTED** [review]
This change is required
https://github.com/hmcts/spring-boot-template/blob/04e7412874900d2de06feacf74896565c606a33f/Dockerfile#L1C1-L2

File: build.gradle [hunk 2]
```diff
@@ -192,14 +195,14 @@ def versions = [
...
+        jacksonDatabind                 : '2.14.2',
         commonsIO                       : '2.8.0',
         commonsValidator                : '1.6',
         restAssured                     : '4.4.0',
-        lombok                          : '1.18.16',
+        lombok                          : '1.18.26',
         springDocOpenUi                 : '1.6.6',
         springSecurityVersion           : '5.7.3',
-        probateCommonsVersion           : '1.4.9',
+        probateCommonsVersion           : 'DTSPB-3524-A-v1.4.10',
```
Comment (DelythJustice) [inline build.gradle]: Just a note to remember this guy is here

File: src/functionalTest/resources/json/bulkScanUpdateCaseExceptionRecordExtendExpiryPA8A.json [hunk 3]
```diff
@@ -111,12 +111,12 @@
             "document_filename":"19365040100100002.pdf",
             "document_binary_url":"http://dm-store:5006/documents/dfd74122-8a21-4bab-b7d1-e6f3d9ca3dad/binary"
           },
-          "scannedDate":"2020-01-02T17:42:03.868Z",
+          "scannedDate":"2020-01-02T17:42:03.868",
```
Comment (DelythJustice) [inline src/functionalTest/resources/json/bulkScanUpdateCaseExceptionRecordExtendExpiryPA8A.json:114]: why this change? do you know if we have Z or not have said in actual?

File: src/functionalTest/resources/json/expectedBulkScanTransformExceptionRecordOutputCombCitizenPA1P.json [hunk 4]
```diff
@@ -1 +1 @@
...lationship":"Son","adoptedInOrOut":"in"}}],"spouseOrPartner":"Yes","applyingAsAnAttorney":"Yes","attorneyOnBehalfOfNameAndAddress":[{"value":{"name":"Neil Kinnock","address":{"AddressLine1":"Not Number 10","AddressLine2":"Seeing Red","AddressLine3":"","County":"Greater London","PostTown":"London Town","PostCode":"NW1 1LD","Country":""}}}],"mentalCapacity":"Yes","courtOfProtection":"No","epaOrLpa":"Yes","epaRegistered":"No","domicilityCountry":"","domicilityIHTCert":"Yes","willDate":"2013-06-20","willsOutsideOfUK":"Yes","willGiftUnderEighteen":"No","notifiedApplicants":"No","foreignAsset":"Yes","foreignAssetEstateValue":"2000000","primaryApplicantSecondPhoneNumber":"701231231231","applicationFeePaperForm":"21500","feeForCopiesPaperForm":"2000","totalFeePaperForm":"23500","paperPaymentMethod":"cheque","paymentReferenceNumberPaperform":"CHK-REF-123456","dateOfDivorcedCPJudicially":"2015-06-01","courtOfDecree":"Manchester","bulkScanCaseReference":"1000002","outsideUKGrantCopies":1,"childrenOverEighteenSurvived":"2","childrenUnderEighteenSurvived":"1","grandChildrenSurvivedUnderEighteen":"0","grandChildrenSurvivedOverEighteen":"2","caseType":"gop","evidenceHandled":"No"}},"warnings":[]}
```
Comment (DelythJustice) [inline src/functionalTest/resources/json/expectedBulkScanTransformExceptionRecordOutputCombCitizenPA1P.json:1]: what was the change needed for this lot?

**Review by DelythJustice — APPROVED** [review]
